core/tauri/scripts/ipc-protocol.js

```diff
@@ -42,7 +42,8 @@
         }
       })
         .then((response) => {
-          const cb = response.ok ? callback : error
+          const cb =
+            response.headers.get('tauri-response') === 'ok' ? callback : error
```

Comment:
should we use `Tauri-Response` to match the header on Rust side (just for consistency, it doesn't matter logically)